lalrpop/src/grammar/parse_tree.rs

```diff
@@ -334,8 +336,8 @@ impl TerminalLiteral {
     /// applies when we are creating the tokenizer anyhow.
     pub fn precedence(&self) -> usize {
         match *self {
-            TerminalLiteral::Quoted(_) => 1,
-            TerminalLiteral::Regex(_) => 0,
+            TerminalLiteral::Quoted(_, p) => p,
+            TerminalLiteral::Regex(_, p) => p,
```

Comment:
So I think how I would prefer to handle this is to remove this method altogether, and instead have `token_check` consult a map to compute the precedence. (As opposed to adding the precedence as a second field in `TerminalLiteral`.) I prefer `TerminalLiteral` to just reflect the bytes being matched; precedence is a concept that only applies when we are using the generated tokenizer.